### PR TITLE
Add support for GitLab, Bitbucket, and Azure DevOps to the Focus View

### PR DESCRIPTION
**Pull request by jdgarcia**
https://gitkraken.atlassian.net/browse/GKCS-5569

Summary:
- Show a "Connect a provider" page when no support provider is currently connected
- Adds support for the cloud versions of GitLab, Bitbucket, and Azure DevOps to the Focus View
- Adds a provider dropdown to the Focus View if you have more than one supported provider connected
- Saves your selected provider to storage

### CODE REVIEW

File: src/shared.ts [hunk 1]
```diff
@@ -107,7 +108,12 @@ export const sessionCachedFetch = async <T>(
 	const sessionStorage = await storage.session.get(key);
 	const data = sessionStorage[key] as CachedFetchResponse<T> | undefined;
 	if (data && data.timestamp > Date.now() - cacheTimeMinutes * 60 * 1000) {
+		debug('Cache hit:', key);
 		return data.data;
+	} else if (data) {
+		debug('Cache stale:', key);
+	} else {
+		debug('Cache miss:', key);
```
Comment (miggy-e) [inline src/shared.ts:116]: I'm good with keeping this in, just wanted to double check that you intended to keep this

**Review by miggy-e — APPROVED** [review]

File: src/shared.ts (hunk 1 above)
Comment (jdgarcia) [reply src/shared.ts:116]: yeah, it doesn't log anything if the mode is production

**Review by gitkrakel — CHANGES_REQUESTED** [review]
From a quick test, GitHub and Bitbucket PRs LGTM.

However when I login to an account without any integrations connected (e.g. an account I created with email and password), the popup is stuck loading:

image

Behavior is consistent on both Chromium and Firefox.

**Comment by jdgarcia** [comment]
@gitkrakel I fixed the issue. I had it working in a previous commit but then accidentally broke it was I was moving around the loading spinner logic.

**Review by gitkrakel — APPROVED** [review]

**Pull request merged**